src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -80,7 +80,13 @@ public abstract class TestResult : ITestResult
         private string _message;
         private string _stackTrace;
 
-        private List<AssertionResult> _assertionResults = new List<AssertionResult>();
+        private readonly List<AssertionResult> _assertionResults = new List<AssertionResult>();
+
+        /// <summary>
+        /// TestAttachments attached to result.
+        /// Only exposed for testing.
+        /// </summary>
+        internal ICollection<TestAttachment> TestAttachments { get; } = new List<TestAttachment>();
```

Comment:
You're right. I had been keeping this as an 'add-only' collection, but it would be more in-keeping with the rest of the TestResult class if this was exposed to the user in a ReadOnlyCollection. I'll expose this list and the TestAttachment type.